src/edu/gatech/cs2340/trydent/TrydentEngine.java

```diff
@@ -0,0 +1,145 @@
+package edu.gatech.cs2340.trydent;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Main class that kicks off the TrydentEngine.
+ * 
+ * As a developer using the TrydentEngine, all you have to do is call
+ * TrydentEngine.start() to get started.
```

Comment:
The trouble with static initializer blocks is that they'd only be activated at the point the JVM loaded in the TrydentEngine class; IE the first time TrydentEngine appears somewhere (directly or indirectly) in the end users' code. Lazy initialization in general has the same problem.

That seems really magic and weird and undesirable. It's running stuff in a separate thread, it needs to be started somehow. 

Also note that we might very well want to make start() take some parameters in the future, to set up the window or something.